tests/disas.rs

```diff
@@ -311,17 +314,122 @@ fn assert_output(
+                    })
+                    .collect::<Vec<_>>())
+            })?,
+            Err(_) => {
+                assert!(matches!(
+                    isa.triple().architecture,
+                    target_lexicon::Architecture::Pulley32 | target_lexicon::Architecture::Pulley64
+                ));
+                disas_elf(&mut actual, &bytes, |bytes, _addr| {
+                    let mut result = vec![];
+
+                    let mut disas = pulley_interpreter::disas::Disassembler::new(bytes);
+                    disas.offsets(false);
+                    disas.hexdump(false);
+                    let mut decoder = pulley_interpreter::decode::Decoder::new();
+
+                    loop {
+                        let addr = disas.bytecode().position();
+
+                        match decoder.decode_one(&mut disas) {
+                            // If we got EOF at the initial position, then we're done disassembling.
+                            Err(pulley_interpreter::decode::DecodingError::UnexpectedEof {
+                                position,
+                            }) if position == addr => break,
+
+                            // Otherwise, propagate the error.
+                            Err(e) => {
+                                return Err(anyhow::Error::from(e))
+                                    .context("failed to disassembly pulley bytecode");
+                            }
+
+                            Ok(()) => {
+                                let inst = disas
+                                    .disas()
+                                    .lines()
+                                    .map(|l| l.trim().to_string())
+                                    .filter(|l| !l.is_empty())
+                                    .next_back()
+                                    .unwrap();
+                                let addr = u64::try_from(addr).unwrap();
+                                let is_jump = inst.contains("jump") || inst.contains("br_if");
+                                let is_ret = inst == "ret";
+                                result.push((addr, is_jump, is_ret, inst));
+                            }
+                        }
+                    }
+
+                    Ok(result)
+                })?
+            }
+        },
     }
     let actual = actual.trim();
     assert_or_bless_output(path, wat, actual)
 }
 
-fn disas_elf(disas: &capstone::Capstone, result: &mut String, elf: &[u8]) -> Result<()> {
-    use capstone::InsnGroupType::{CS_GRP_JUMP, CS_GRP_RET};
+fn disas_elf<I>(
+    result: &mut String,
+    elf: &[u8],
+    disas: impl Fn(&[u8], u64) -> Result<I>,
+) -> Result<()>
+where
+    I: IntoIterator<
+        Item = (
+            // Address
+            u64,
+            // is jump?
+            bool,
+            // is return?
+            bool,
+            // disasembly
+            String,
+        ),
```

Comment:
Mind switching this to something like `struct Instr { /* named fields here */ }`?